### PR TITLE
(MODULES-2207) Modulesync

### PR DESCRIPTION
**Pull request by ferventcoder**
This brings the following items under modulesync:
* .gitignore
* .travis.yml
* CONTRIBUTING.md
* Gemfile
* appveyor.yml
* spec/spec.opts

More importantly, this ensures that acl has up to date Gemfile, Travis CI and
AppVeyor configurations that allow it to be tested properly.

### CODE REVIEW

File: tests/init.pp [hunk 1]
```diff
@@ -53,12 +53,12 @@
   permissions                => [
     { identity    => 'Administrator',
       rights      => ['full'],
-      type        => 'allow',
+      'type'      => 'allow',
```
Comment (cyberious) [inline tests/init.pp]: how did this sneak into this PR?

Comment (cyberious) [inline tests/init.pp]: Also this was updated to perm_type as part of the future parse fixes.

Comment (ferventcoder) [inline tests/init.pp]: This is the second commit?

Comment (ferventcoder) [inline tests/init.pp]: I'll fix it up.

**Comment by Iristyle** [comment]
FYI @ferventcoder on 6b419ff the commit message doesn't match what you did

**Comment by ferventcoder** [comment]
@Iristyle good catch - that's what I originally did. Will fix it up.

**Comment by ferventcoder** [comment]
All fixed up and waiting on spec validation.

**Comment by ferventcoder** [comment]
Ready to go

**Pull request merged**